micrometer-registry-graphite/runtime/src/main/java/io/quarkiverse/micrometer/registry/graphite/GraphiteMeterRegistryProvider.java

```diff
@@ -0,0 +1,71 @@
+package io.quarkiverse.micrometer.registry.graphite;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+import javax.enterprise.inject.Instance;
+import javax.enterprise.inject.Produces;
+import javax.inject.Singleton;
+
+import org.eclipse.microprofile.config.Config;
+
+import io.micrometer.core.instrument.Clock;
+import io.micrometer.core.instrument.util.HierarchicalNameMapper;
+import io.micrometer.graphite.GraphiteConfig;
+import io.micrometer.graphite.GraphiteDimensionalNameMapper;
+import io.micrometer.graphite.GraphiteHierarchicalNameMapper;
+import io.micrometer.graphite.GraphiteMeterRegistry;
+import io.quarkus.arc.DefaultBean;
+import io.quarkus.micrometer.runtime.export.ConfigAdapter;
+
+@Singleton
+public class GraphiteMeterRegistryProvider {
+    static final String PREFIX = "quarkus.micrometer.export.graphite.";
+
+    @Produces
+    @Singleton
+    @DefaultBean
+    public GraphiteConfig configure(Config config) {
+        final Map<String, String> properties = ConfigAdapter.captureProperties(config, PREFIX);
+        return ConfigAdapter.validate(properties::get);
+    }
+
+    @Produces
+    @Singleton
+    @DefaultBean
+    @GraphiteNameMapper
+    public HierarchicalNameMapper nameMapper(GraphiteConfig config) {
+        return config.graphiteTagsEnabled() ? new GraphiteDimensionalNameMapper()
+                : new GraphiteHierarchicalNameMapper(config.tagsAsPrefix());
+    }
+
+    @Produces
+    @Singleton
+    @DefaultBean
+    public GraphiteMeterRegistry registry(Instance<GraphiteMeterRegistryConfigCustomizer> customizers,
+            GraphiteConfig config,
+            @GraphiteNameMapper HierarchicalNameMapper nameMapper,
+            Clock clock) {
+        GraphiteMeterRegistry graphiteMeterRegistry = new GraphiteMeterRegistry(config, clock, nameMapper);
+
+        List<GraphiteMeterRegistryConfigCustomizer> sortedCustomizers = sortCustomizersInDescendingPriorityOrder(customizers);
+
+        for (GraphiteMeterRegistryConfigCustomizer customizer : sortedCustomizers) {
+            customizer.customize(graphiteMeterRegistry.config());
```

Comment:
I had been hoping to avoid customizers by sticking with MeterFilter/MeterFilterConstraint, and then custom registry construction using injected config. I know Spring uses them, but I think that is a result of how the autoconfig mechanisms work (such that customizers are easier). That complexity doesn't exist here...

I would rather make the default easier to suppress/replace (I have a plan/open issue for that, as there is a weakness now) than add customizers, unless you can make a strong argument otherwise.

Reply:
Yup, makes sense =]